src/shared_modules/content_manager/src/actionOrchestrator.hpp

```diff
@@ -76,8 +76,8 @@ class ActionOrchestrator final
             // If the database exists, get the last offset
             if (m_spBaseContext->spRocksDB)
             {
-                const auto value = m_spBaseContext->spRocksDB->getLastKeyValue().second;
-                spUpdaterContext->currentOffset = value.empty() ? 0 : std::stoi(value.ToString());
+                spUpdaterContext->currentOffset =
+                    std::stoi(m_spBaseContext->spRocksDB->getLastKeyValue().second.ToString());
```

Comment:
Question: Isn't this handled on the ExecutionContext class?

Reply:
No. The ExecutionContext stage is executed at the beginning of the process and it just ensures that the correct offset is set on the database. Note that `currentOffset` is not modified in the execution context stage.

Then the orchestration is executed plenty of times within a process, dispatched by a schedule (or on-demand), and each time it is dispatched, the offset may change (see [CtiAPIDownloader](https://github.com/wazuh/wazuh/blob/04b16d9c0bd49b79e2fceccb1d19c4ad1b25c9d9/src/shared_modules/content_manager/src/components/CtiApiDownloader.hpp#L26) class) so it has to be read each time the orchestration starts.

Reply:
Awesome, didn't knew the offset could change in between